pkg/plugins/runtime/gateway/testdata/tls/tcp-route.yaml

```diff
@@ -45,6 +45,13 @@ Clusters:
                 resourceApiVersion: V3
           sni: api-service{mesh=default}
       type: EDS
+      typedExtensionProtocolOptions:
+        envoy.extensions.upstreams.http.v3.HttpProtocolOptions:
+          '@type': type.googleapis.com/envoy.extensions.upstreams.http.v3.HttpProtocolOptions
+          commonHttpProtocolOptions:
+            idleTimeout: 3600s
+          explicitHttpConfig:
+            httpProtocolOptions: {}
```

Comment:
It's because we are building map of protocol in mesh context based on EndpointMap. Test EndpointMap has a definition of this service with a `kuma.io/protocol: http`

Reply:
Should we fix the test case?

Reply:
The difference is that previously we were taking `tags` from the destination field in the policy and putting them into the endpoints structure. Based on these endpoints control-plane was getting the protocol. Because in the destination there is no tag with the policy only the service name we were getting the TCP. Now I've changed that instead of creating endpoints based on the tags in the destinations we are retrieving the data from the protocol map. It has HTTP protocol tag so that's why we are getting it now.

Reply:
Maybe to keep it constant I should take the value of the protocol tag from the `destination`?